ha/core/system_health/bootstrap/json_kv_convertor.py

```diff
@@ -43,85 +52,158 @@ def __init__(self):
             self._uid = self._last_modified = self._status = self._key = None
 
-
-    def _update_kv(self, k , v):
-
-        kv = f"{k}:{v}\n"
-
-        for pattern in self._filter_list:
-            if re.search(pattern, kv):
-                if pattern == "uid:":
-                    if re.search("health.specifics", kv) or re.search("sites.uid:", kv) or re.search("sites.racks.uid:", kv) or  re.search("sites.racks.cortx_nodes.uid:", kv) :
-                        break
-                elif  re.search("sites.racks.cortx_nodes.last_updated:", kv):
-                    break
-
-                self._update_health(k,v)
-
-
-    def _walk(self, d):
-
-        for k,v in d.items():
-            if isinstance(v, dict):
-                self._path.append(k)
-                self._walk(v)
-                self._path.pop()
-
-            elif isinstance(v, list):
-                self._path.append(k)
-                if not v:
-                    key = (".".join(self._path))
-                    val = v
-                    self._update_kv(key, val)
-                    self._path.pop()
-
-                else:
-                    if isinstance(v[0], dict):
-                        for v_int in v:
-                            self._walk(v_int)
-                    # End value in the KV  is a list
-                    else:
-                        key = (".".join(self._path))
-                        val = v
-                        self._update_kv(key, val)
-                    self._path.pop()
-
-            else:
-                # conditions handled isinstance(v, str) or isinstance(v, int) etc
-                self._path.append(k)
-                key = (".".join(self._path))
-                val = v
-                self._update_kv(key, val)
-                self._path.pop()
+    def _get_required_compute_kv(self, key=None):
+        """
+           The node health json structure is:
+           {
+              node: {
+                        compute: []
+                    },
+                    {
+                        storage: []
+                    }
+           }
+           This routine will help to get the all required key values from
+           confstore. Prefix: 'node_health' key: node>compute[0]
+           uid, last_updated_status and health status will be updated for
+           each compute as well as sub component of compute
+           Ex: node.compute.hw.cpu.uid CPU-0
+        """
+        for compute_res in self.compute_resource_list:
+            if isinstance(self.compute_health_list[compute_res], dict):
+                for comp in self.compute_health_list[compute_res]:
+                    new_key = key + f'.{compute_res}.{comp}'
+                    self._parse_health_comp_dict(self.compute_health_list[compute_res][comp], new_key)
+
+    def _get_required_storage_kv(self, key: str = None) -> None:
```

Comment:
Pls verify that this creates keys for all components and subcomponents inside storage

Reply:
Yes. I have checked and mostly I verified it also. If you can see anything, we can connect again.